Mage.Sets/src/mage/cards/w/WeepingAngel.java

```diff
@@ -0,0 +1,156 @@
+package mage.cards.w;
+
+import java.util.UUID;
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.common.SpellCastOpponentTriggeredAbility;
+import mage.abilities.effects.ContinuousEffectImpl;
+import mage.abilities.effects.ReplacementEffectImpl;
+import mage.cards.Card;
+import mage.constants.*;
+import mage.abilities.keyword.FlashAbility;
+import mage.abilities.keyword.FirstStrikeAbility;
+import mage.abilities.keyword.VigilanceAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.game.events.DamagePermanentEvent;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+
+/**
+ *
+ * @author jimga150
+ */
+public final class WeepingAngel extends CardImpl {
+
+    public WeepingAngel(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT, CardType.CREATURE}, "{1}{U}{B}");
+        
+        this.subtype.add(SubType.ALIEN);
+        this.subtype.add(SubType.ANGEL);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(2);
+
+        // Flash
+        this.addAbility(FlashAbility.getInstance());
+
+        // First strike
+        this.addAbility(FirstStrikeAbility.getInstance());
+
+        // Vigilance
+        this.addAbility(VigilanceAbility.getInstance());
+
+        // Whenever an opponent casts a creature spell, Weeping Angel isn't a creature until end of turn.
+        this.addAbility(new SpellCastOpponentTriggeredAbility(
+                new WeepingAngelMarbleizeEffect(), StaticFilters.FILTER_SPELL_A_CREATURE, false
+        ));
+
+        // If Weeping Angel would deal combat damage to a creature, prevent that damage and that creature's owner shuffles it into their library.
+        this.addAbility(new SimpleStaticAbility(new WeepingAngelDamageEffect()));
+    }
+
+    private WeepingAngel(final WeepingAngel card) {
+        super(card);
+    }
+
+    @Override
+    public WeepingAngel copy() {
+        return new WeepingAngel(this);
+    }
+}
+
+// Adapted from LoseCreatureTypeSourceEffect
+class WeepingAngelMarbleizeEffect extends ContinuousEffectImpl {
+    
+    public WeepingAngelMarbleizeEffect() {
```

Comment:
remember for custom classes, package private constructor and private copy constructor